packages/react-components/react-tree/src/hooks/useRovingTabIndexes.ts

```diff
@@ -1,13 +1,27 @@
 import * as React from 'react';
 import { HTMLElementWalker } from '../utils/createHTMLElementWalker';
+import { useListenFocusedElement } from '@fluentui/react-tabster';
+import { elementContains } from '@fluentui/react-utilities';
 
 /**
  * https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_roving_tabindex
  */
 export function useRovingTabIndex() {
   const currentElementRef = React.useRef<HTMLElement>();
+  const walkerRef = React.useRef<HTMLElementWalker | null>(null);
+  useListenFocusedElement(element => {
+    if (
+      walkerRef.current &&
+      element &&
+      elementContains(walkerRef.current.root, element) &&
+      element.getAttribute('role') === 'treeitem'
```

Comment:
nit: let's re-arrange checks, so `element.getAttribute('role') === 'treeitem'` will be first as it's significantly cheaper one